### ISSUE TITLE
Apple Health is MASSIVE and draining my battery

### ISSUE THREAD

**Issue by drspike** [Artificial-Pancreas/iAPS#703]
Is there any way to optimize Apple Health with iAPS so the file isn’t so large (12 gb with 4 gb of “insulin” alone) and Apple health uses 44% of my battery in a day.

**Comment by itsmojo** [comment]
I don’t think this is an iAPS specific problem. I had been running Trio for a couple of weeks when I starting noticing Health taking a huge percentage of my CPU after my Heath data had gotten to 16GB with over 6GB of Insulin Delivery data. I noticed that my Apple Health DB size was growing by 100’s of MB in minutes even after I turned off Trio Health updates. Prior to this I had been running Loop dev for several weeks and prior to that I had been running iAPS.

To recover I turned off Heath iCloud backups (fyi, my iCloud backup size for Health data was “only” 2G), completely wiped my phone, and then recovered everything (except Heath) from my backups. Now my iPhone battery isn’t getting run down excessive fast & Health DB isn’t growing by leaps and bounds.

![IMG_5172](https://github.com/Artificial-Pancreas/iAPS/assets/23020111/a8b5f840-7f4c-4d08-8b98-3097b57ee044)
![IMG_5177](https://github.com/Artificial-Pancreas/iAPS/assets/23020111/25eb9e9e-9677-4f40-a029-229d92664cc1)
![IMG_5178](https://github.com/Artificial-Pancreas/iAPS/assets/23020111/29e1af4e-784e-4498-816b-bcc835420922)
